packages/flutter/lib/src/services/text_input.dart

```diff
@@ -946,4 +928,100 @@ class TextInput {
     }());
     return true;
   }
+
+  MethodChannel _channel;
+
+  TextInputConnection _currentConnection;
+  TextInputConfiguration _currentConfiguration;
+  TextEditingValue _currentTextEditingValue;
+
+  Future<dynamic> _handleTextInputInvocation(MethodCall methodCall) async {
+    if (_currentConnection == null)
+      return;
+    final String method = methodCall.method;
+
+    // The requestExistingInputState request needs to be handled regardless of
+    // the client ID, as long as we have a _currentConnection.
+    if (method == 'TextInputClient.requestExistingInputState') {
+      assert(_currentConnection._client != null);
+      _attach(_currentConnection, _currentConfiguration);
+      // This will be null if we've never had a call to [_setEditingState].
+      if (_currentTextEditingValue != null) {
+        _setEditingState(_currentTextEditingValue);
+      }
```

Comment:
It turns out these lines never get called because of https://github.com/flutter/flutter/issues/47137 - see https://github.com/flutter/flutter/pull/47177

Reply:
Quick question: in Android or in general?

In Flutter for Web, this is the main part that we are syncing the state from Framework to Platform. It works for the web engine so far.

Let's debug and try to see why it is not the case for Android.

Reply:
Android is the only platform that would go down this codepath today.  It shouldn't be a problem for web.

See the linked PR for more details - I don't think this is a problem for web, but it is a problem for Android where the engine side loses state.